src/uuid/uuidbuilder.rs

```diff
@@ -33,174 +32,196 @@ pub struct UUIDBuilder {
+    /// called from. The `rand_b` portion is what uniquely identifies this uE.
     ///
-    /// ```rust
-    /// use up_rust::UUIDBuilder;
+    /// # Returns
     ///
-    /// let builder = UUIDBuilder::default();
-    /// let uuid = builder.build();
-    /// assert!(uuid.is_uprotocol_uuid());
-    /// ```
-    fn default() -> Self {
-        Self::new()
+    /// UUID with consistent `rand_b` portion, which uniquely identifies this uE
+    pub fn build() -> UUID {
+        #[allow(clippy::redundant_closure)]
+        static UUIDBUILDER_SINGLETON: Lazy<UUIDBuilder> = Lazy::new(|| UUIDBuilder::new());
+        UUIDBUILDER_SINGLETON.build_from_instance()
     }
-}
 
-impl UUIDBuilder {
     /// Creates a new builder for creating uProtocol UUIDs.
     ///
-    /// The same bulder instance can be used to create one or more UUIDs
-    /// by means of invoking [`UUIDBuilder::build`].
+    /// The same builder instance can be used to create one or more UUIDs
+    /// by means of invoking [`UUIDBuilder::build_from_instance`].
     ///
-    /// # Examples
+    /// # Note
     ///
-    /// ```rust
-    /// use up_rust::UUIDBuilder;
-    ///
-    /// let builder = UUIDBuilder::new();
-    /// let uuid1 = builder.build();
-    /// assert!(uuid1.is_uprotocol_uuid());
-    /// let uuid2 = builder.build();
-    /// assert!(uuid2.is_uprotocol_uuid());
-    /// assert_ne!(uuid1, uuid2);
-    /// ```
-    pub fn new() -> Self {
+    /// For internal testing purposes only. For end-users, please use [`UUIDBuilder::build()`]
+    pub(crate) fn new() -> Self {
         UUIDBuilder {
             msb: AtomicU64::new(0),
-            // set variant to RFC4122
-            lsb: rand::thread_rng().gen::<u64>() & BITMASK_CLEAR_VARIANT
-                | crate::uuid::VARIANT_RFC4122,
+            lsb: random::<u64>() & BITMASK_CLEAR_VARIANT | crate::uuid::VARIANT_RFC4122,
         }
     }
 
     /// Creates a new UUID for the current system time.
     ///
-    /// # Panics
-    ///
-    /// if the system time is either
-    /// * set to a point in time before UNIX Epoch, or
-    /// * set to a point in time later than UNIX Epoch + 0xFFFFFFFFFFFF seconds
-    ///
-    /// # Examples
-    ///
-    /// ```rust
-    /// use up_rust::UUIDBuilder;
+    /// # Note
     ///
-    /// let uuid = UUIDBuilder::new().build();
-    /// assert!(uuid.is_uprotocol_uuid());
-    /// assert!(uuid.get_time().is_some());
-    /// ```
-    pub fn build(&self) -> UUID {
-        let now = SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .expect("current system time is set to a point in time before UNIX Epoch");
-        let now_millis = u64::try_from(now.as_millis())
-            .expect("current system time is set to a point in time too far in the future");
-        self.build_with_instant(now_millis)
+    /// For internal testing purposes only. For end-users, please use [`UUIDBuilder::build()`]
+    pub(crate) fn build_from_instance(&self) -> UUID {
```

Comment:
How about remove the `build_internal` and only leave `build_from_instance`? Or do you plan to add something in `build_from_instance` in the future?

Reply:
I think this is reasonable. :+1:  I'm already doing a superset of the one remaining test.